text/0000-cfg-target.md

```diff
@@ -0,0 +1,99 @@
+- Feature Name: `cfg-target`
+- Start Date: 2020-09-27
+- RFC PR: [rust-lang/rfcs#2991](https://github.com/rust-lang/rfcs/pull/2991)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+[summary]: #summary
+
+This proposes a new `cfg`: `target`, which matches the entire target triple
+string (e.g. `arm-unknown-linux-gnueabihf`). This also adds a `CARGO_CFG_TARGET`
+environment variable for parity with other `CARGO_CFG_*` variables.
+
+# Motivation
+[motivation]: #motivation
+
+To `#[cfg]` against a specific target, a `build.rs` script is required to emit a
+custom `cfg` based on the `TARGET` environment variable. Adding a build script
+increases compile time and makes a crate incompatible with certain build
+systems.
+
+Otherwise, all available components would need to be specified separately:
+`target_arch`, `target_vendor`, `target_os`, and `target_env`. This can be very
+cumbersome. Note that the target ABI cannot currently be `#[cfg]`-ed against, so
+a `build.rs` is still necessary to match all target components.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+This would act like existing `target_*` configurations but match against all
+components (except `target_feature`).
+
+```rust
+#[cfg(target = "x86_64-apple-ios-macabi")]
+mod mac_catalyst;
+```
+
+This includes `#[cfg_attr(target = "...", attr)]`.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+`target` is a key-value option set once with the target's Rust triple.
+
+Example values:
+
+- `"aarch64-apple-darwin"`
+- `"arm-unknown-linux-gnueabihf"`
+- `"x86_64-apple-ios-macabi"`
+- `"x86_64-pc-windows-gnu"`
+- `"x86_64-pc-windows-msvc"`
+- `"x86_64-unknown-linux-gnu"`
+
+# Drawbacks
```

Comment:
Another drawback might be that technically this could be a breaking change with anybody that conditionally adds a `--cfg target=...` cfg either via RUSTFLAGS or `	cargo:rustc-cfg=target=...` from a build script.

This goes for your `target_abi` PR also.

Reply:
Right, so if a person's [build.rs](https://github.com/Lokathor/fermium/blob/main/build.rs) had lines like say
```rust
  let target = env::var("TARGET").expect("Couldn't read `TARGET`");
  println!("cargo:rustc-env=TARGET={}", target);
```
Then technically they'd hit a small break ;P

Reply:
No, that should be fine. The code would have to be:
```rust
let target = env::var("TARGET").expect("Couldn't read `TARGET`");
println!("cargo:rustc-cfg=target={}", target);
```
in order to have a break.

Reply:
The string is equal to the `TARGET` environment var anyway, so `target` would not change.

Reply:
Yeah, that's what I meant by:

> Actually, this would define the target identically to what it was before, so they'd have to define it as something different (unless their definition would take priority?), or only in some cases, I guess.

It might be a breaking change for:

```rust
let target = env::var("CARGO_CFG_TARGET_ARCH").expect("Couldn't read `CARGO_CFG_TARGET_ARCH`");
println!("cargo:rustc-cfg=target={}", target);
```

or

```rust
if some_condition {
    let target = env::var("TARGET").expect("Couldn't read `TARGET`");
    println!("cargo:rustc-cfg=target={}", target);
}
```

But these are pretty obscure cases...